chassis/core/src/main/java/com/griddynamics/jagger/engine/e1/aggregator/workload/DurationLogProcessor.java

```diff
@@ -206,40 +212,18 @@ public StatisticsGenerator generate() throws IOException {
             int currentCount = 0;
             int totalCount = 0;
             int extendedInterval = intervalSize;
-            StatisticsCalculator windowStatisticsCalc = new StatisticsCalculator();
-            StatisticsCalculator globalStatisticsCalc = new StatisticsCalculator();
-
-            MetricDescriptionEntity throughputDescription = persistMetricDescription(THROUGHPUT_ID, THROUGHPUT_TPS, taskData);
-            MetricDescriptionEntity latencyDesc = persistMetricDescription(LATENCY_ID, LATENCY_SEC, taskData);
-            MetricDescriptionEntity latencyStdDevDesc = persistMetricDescription(LATENCY_STD_DEV_ID, LATENCY_STD_DEV_SEC, taskData);
-
-            Map<Double, MetricDescriptionEntity> percentileMap = new HashMap<>(getTimeWindowPercentilesKeys().size());
-            for (Double percentileKey : getTimeWindowPercentilesKeys()) {
-                String metricStr = StandardMetricsNamesUtil.getLatencyMetricName(percentileKey, false);
-                percentileMap.put(percentileKey, persistMetricDescription(metricStr, metricStr, taskData));
-            }
-
-            // collect-aggregate plot data
-            LogReader.FileReader<DurationLogEntry> fileReader = null;
-            try {
-                fileReader = logReader.read(path, DurationLogEntry.class);
+            int addedStatistics = 0;
+            try (LogReader.FileReader<DurationLogEntry> fileReader = logReader.read(path, DurationLogEntry.class)) {
                 for (DurationLogEntry logEntry : fileReader) {
                     log.debug("Log entry {} time", logEntry.getTime());
-
                     while (logEntry.getTime() > currentInterval) {
-                        log.debug("processing count {} interval {}", currentCount, intervalSize);
-
+                        log.debug("Processing count {} interval {}", currentCount, intervalSize);
                         if (currentCount > 0) {
                             double throughput = (double) currentCount * 1000 / extendedInterval;
                             long currentTime = time - extendedInterval / 2;
-                            statistics.add(new MetricPointEntity(currentTime, throughput, throughputDescription));
-                            statistics.add(new MetricPointEntity(currentTime, windowStatisticsCalc.getMean() / 1000, latencyDesc));
-                            statistics.add(new MetricPointEntity(currentTime, windowStatisticsCalc.getStandardDeviation() / 1000, latencyStdDevDesc));
-
-                            for (Double percentileKey : getTimeWindowPercentilesKeys()) {
-                                Double percentileValue = windowStatisticsCalc.getPercentile(percentileKey) / 1000D;
-                                statistics.add(new MetricPointEntity(time, percentileValue, percentileMap.get(percentileKey)));
-                            }
+                            // first point is removed because it's value very high and it breaks statistics
```

Comment:
I haven't found the root cause of this problem yet. First points are added with high values, so the problem is not in aggregation, but in collection data stage.